Packages/Tracking Preview/Input Action Hands/LeapHandInput.cs

```diff
@@ -0,0 +1,171 @@
+    //    public AxisControl width { get; private set; }
+
+
+    //    protected override void FinishSetup()
+    //    {
+    //        parentBoneIndex = GetChildControl<IntegerControl>("parentBoneIndex");
+    //        position = GetChildControl<Vector3Control>("position");
+    //        rotation = GetChildControl<QuaternionControl>("rotation");
+    //        length = GetChildControl<AxisControl>("length");
+    //        width = GetChildControl<AxisControl>("width");
+
+    //        base.FinishSetup();
+    //    }
+
+    //    public override unsafe LeapBone ReadUnprocessedValueFromState(void* statePtr)
+    //    {
+    //        return new LeapBone()
+    //        {
+    //            parentBoneIndex = (uint)parentBoneIndex.ReadUnprocessedValueFromState(statePtr),
+    //            position = position.ReadUnprocessedValueFromState(statePtr),
+    //            rotation = rotation.ReadUnprocessedValueFromState(statePtr),
+    //            length = length.ReadUnprocessedValueFromState(statePtr),
+    //            width = width.ReadUnprocessedValueFromState(statePtr)
+    //        };
+    //    }
+
+    //    public override unsafe void WriteValueIntoState(LeapBone value, void* statePtr)
+    //    {
+    //        parentBoneIndex.WriteValueIntoState((int)value.parentBoneIndex, statePtr);
+    //        position.WriteValueIntoState(value.position, statePtr);
+    //        rotation.WriteValueIntoState(value.rotation, statePtr);
+    //        length.WriteValueIntoState(value.length, statePtr);
+    //        width.WriteValueIntoState(value.width, statePtr);
+    //    }
+    //}
+
+    #endregion
+
+    /// <summary>
+    /// The structure used to pass hand data to InputDevices
+    /// </summary>
+    public struct LeapHandState : IInputStateTypeInfo
+{
+    public FourCC format => new FourCC('L', 'H', 'N', 'D');
+
+    #region Tracking ID
+
+    [InputControl(name = "Tracked", layout = "Integer")]
+    public int tracked;
+
+    #endregion
+
+    #region Palm
+
+    [InputControl(name = "Palm Position", layout = "Vector3")]
+    public Vector3 position;
+
+    [InputControl(name = "Palm Direction", layout = "Quaternion")]
+    public Quaternion direction;
+
+    #endregion
+
+    #region Index
+
+    [InputControl(name = "Index Tip Position", layout = "Vector3")]
+    public Vector3 indexTipPosition;
+
+    #endregion
+
+    #region Extras
+
+    [InputControl(name = "Is Pinching", layout = "Button")]
+    public float isPinching;
+
+    [InputControl(name = "Is Grabbing", layout = "Button")]
+    public float isGrabbing;
+
+    #endregion
+}
+    /// <summary>
+    /// An Input Device generated using two individual hand Input Devices representing left and right hands.
+    /// </summary>
+#if UNITY_EDITOR
+    [InitializeOnLoad]
+#endif
+    [InputControlLayout(displayName = "Ultraleap", stateType = typeof(LeapHandState), commonUsages = new[] { "LeftHand", "RightHand" }, canRunInBackground = true, isNoisy = true)]
```

Comment:
The isNoisy property requires us to specify a minimum version for the com.unity.inputsystem on the asmdef - I think at least 1.4 (possibly 1.4.4). It's missing from 1.3.0 which is what I see when using this code.

Reply:
https://docs.unity3d.com/Packages/com.unity.inputsystem@1.4/api/UnityEngine.InputSystem.Layouts.InputControlLayoutAttribute.html